Source/NSGridView.m

```diff
@@ -0,0 +1,511 @@
+/* Implementation of class NSGridView
+   Copyright (C) 2020 Free Software Foundation, Inc.
+   
+   By: Gregory John Casamento
+   Date: 08-08-2020
+
+   This file is part of the GNUstep Library.
+   
+   This library is free software; you can redistribute it and/or
+   modify it under the terms of the GNU Lesser General Public
+   License as published by the Free Software Foundation; either
+   version 2.1 of the License, or (at your option) any later version.
+   
+   This library is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+   Lesser General Public License for more details.
+   
+   You should have received a copy of the GNU Lesser General Public
+   License along with this library; if not, write to the Free
+   Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+   Boston, MA 02110 USA.
+*/
+
+#import <Foundation/NSArray.h>
+#import "AppKit/NSGridView.h"
+
+#import "GSFastEnumeration.h"
+
+@interface NSGridRow (Private)
+- (void) _setRow: (NSMutableArray *)row;
+- (NSMutableArray *) _row;
+@end
+
+@interface NSGridColumn (Private)
+- (void) _setColumn: (NSMutableArray *)col;
+- (NSMutableArray *) _column;
+@end
+
+@implementation NSGridView
+
+- (instancetype) initWithFrame: (NSRect)frameRect
+{
+  self = [super initWithFrame: frameRect];
+  if (self != nil)
+    {
+      _rows = [[NSMutableArray alloc] initWithCapacity: 10];
+    }
+  return self;
+}
+
+- (instancetype) initWithViews: (NSArray *)rows
+{
+  self = [self initWithFrame: NSZeroRect];
+
+  if (self != nil)
+    {
+      NSMutableArray *mutableRows = [rows mutableCopy];
```

Comment:
The mutable copy here is completely useless. You might need it two lines below, where you add to _rows.

Reply:
I will fix this.  Forgive my lag in response lately, I have taken on a new contract and it's absorbing some of my extra time.  Thanks. GC  I will get back to this sometime later today.

Reply:
Removed.